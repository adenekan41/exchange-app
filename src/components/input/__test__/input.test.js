/* eslint-disable no-undef */
/* -------------------------- External Dependecies -------------------------- */

import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

/* --------------------------- Internal Dependency -------------------------- */
import Input from '..';

const InputMock = ({ type = 'text' }) => {
	return (
		<Input
			id="input"
			type={type}
			placeholder="Enter Text Here"
			label="text"
			value="12345"
			onChange={(e) => e.target.value}
		/>
	);
};
describe('Input Component', () => {
	afterEach(cleanup);

	it('Input Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<InputMock />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Input Renders Input Correctly With Attributes', () => {
		const { getByTestId } = render(<InputMock />);
		expect(getByTestId('input')).not.toHaveAttribute('disabled');
	});

	it('Input matches snapshot', () => {
		const tree = renderer.create(<InputMock />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Input matches snapshot when type is number', () => {
		const tree = renderer.create(<InputMock type="number" />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	// it('Test click event', () => {
	// 	const component = renderer.create(
	// 		<Input
	// 			id="input"
	// 			type="password"
	// 			placeholder="Enter Text Here"
	// 			label="text"
	// 			isInvalid
	// 			value=""
	// 		/>
	// 	);
	// 	renderer.act(() => {
	// 		component.root.findByType('').props.onClick();
	// 	});
	// });
});
