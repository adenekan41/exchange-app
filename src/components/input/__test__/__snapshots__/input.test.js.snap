// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Input matches snapshot 1`] = `
<div
  className="sc-AxjAm gwhofU"
  data-testid="input"
>
  <label
    htmlFor="text"
  >
    text
  </label>
  <input
    className="form-control"
    data-testid="inputs"
    id="input"
    onChange={[Function]}
    placeholder="Enter Text Here"
    type="text"
    value="12345"
  />
</div>
`;

exports[`Input Component Input matches snapshot when type is number 1`] = `
<div
  className="sc-AxjAm gwhofU"
  data-testid="input"
>
  <label
    htmlFor="text"
  >
    text
  </label>
  <input
    className="form-control"
    data-testid="inputs"
    id="input"
    onChange={[Function]}
    placeholder="Enter Text Here"
    type="number"
    value="12345"
  />
</div>
`;
